homeassistant/components/sensor/jewish_calendar.py

```diff
@@ -124,18 +146,64 @@ def state(self):
         date = hdate.HDate(
             today, diaspora=self.diaspora, hebrew=self._hebrew)
 
+        location = hdate.Location(latitude=self.latitude,
+                                  longitude=self.longitude,
+                                  timezone=self.timezone,
+                                  diaspora=self.diaspora)
+
         if self.type == 'date':
             self._state = date.hebrew_date
         elif self.type == 'weekly_portion':
-            self._state = date.parasha
+            # Compute the weekly portion based on the upcoming shabbat.
+            self._state = date.upcoming_shabbat.parasha
         elif self.type == 'holiday_name':
             self._state = date.holiday_description
         elif self.type == 'holyness':
             self._state = date.holiday_type
+        elif self.type == 'upcoming_shabbat_candle_lighting' or \
+                self.type == 'upcoming_candle_lighting':
+            # Start out with Shabbat/Friday.
```

Comment:
The current sensor contains no calculations. All calculations are performed inside hdate. So these calculations should be moved into hdate too.

Reply:
@arigilder I think it does make sense. We want as little as possible of logic in the home assistant project. It removes the need to maintain code which is not directly related to home automation/IOT/... over time.

@balloob  Would it be ok to merge this and move the logic into hdate on our next submission like we did with the original release of the component? Or would you prefer to keep this PR pending while we rework the changes into hdate?

Reply:
Let's update hdate first.

Reply:
@arigilder Please use hdate 0.8.7, it adds all of your changes.